plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -0,0 +1,275 @@
+            except:
+                self.log("Unable to download snapshot: {0}".format(snapshot_url), xbmc.LOGERROR)
+                snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        else:
+            snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        self.log("  snapshot_file name = {0}".format(snapshot_file), xbmc.LOGDEBUG)
+        return snapshot_file
+
+    def stop_camera(self):
+        pass
+
+    def play_camera(self, device_id):
+        self.log("BEGIN play_camera({})".format(device_id), xbmc.LOGNOTICE)
+
+        # Send the command to start the stream and return the stream url.
+        camera = self._get_camera(device_id)
+        stream_url = self.arlo.StartStream(self.basestation, camera)
+
+        # Create a playable item with a path to play.
+        play_item = xbmcgui.ListItem(path=stream_url, label=camera['deviceName'])
+        play_item.setProperty("isPlayable", "true")
+        # Pass the item to the Kodi player.
+        xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+
+        wait_secs = 5
+        is_playing = False
+        timed_out = False
+        start_time = time.time()
+        while not (is_playing or timed_out):
+            is_playing = xbmc.Player().isPlaying()
+            if is_playing:
+                self.log("Player has started...", xbmc.LOGDEBUG)
+            else:
+                if time.time() - start_time < wait_secs:
+                    self.log("Waiting for player to start..", xbmc.LOGDEBUG)
+                    time.sleep(1)
+                else:
+                    self.log("Timed out waiting for player to start", xbmc.LOGWARNING)
+                    timed_out = True
+
+        while is_playing:
+            self.log("Player is playing...", xbmc.LOGDEBUG)
+            time.sleep(1)
+            is_playing = xbmc.Player().isPlaying()
+            if not is_playing:
+                self.log("Player has stopped.", xbmc.LOGDEBUG)
+
+        self._update_arlo_cameras_details()
+
+        self.log("END   play_camera({})".format(device_id), xbmc.LOGNOTICE)
+
+    def arlo_login(self):
+        self.log("BEGIN arlo_login()", xbmc.LOGDEBUG)
+        user_name = REAL_SETTINGS.getSetting('userid')
+        password = REAL_SETTINGS.getSetting('password')
+        self.arlo = resources.lib.Arlo.Arlo(user_name, password)
+        self.basestation = self.arlo.GetDevices('basestation')[0]
+        if self.debug_mode:
+            self.log(json.dumps(self.basestation, indent=4), xbmc.LOGNOTICE)
+        self.log("END   arlo_login()", xbmc.LOGDEBUG)
+
+    def arlo_logout(self):
+        self.log("BEGIN arlo_logout()", xbmc.LOGDEBUG)
+        self.arlo.Logout()
+        self.log("END   arlo_logout()", xbmc.LOGDEBUG)
+
+    def run(self):
+        self.log("BEGIN run()", xbmc.LOGNOTICE)
+        self.log("  arg[0]  : {}".format(self._args[0]), xbmc.LOGNOTICE)
+        self.log("  arg[1]  : {}".format(self._args[2]), xbmc.LOGNOTICE)
```

Comment:
logspam, please remove

Reply:
If you enable debugging in Kodi, the same info is already loged from Kodi itself (`StartScript - calling plugin ...`).

Reply:
Removed for the next pull request